src/mergeForEach.ts

```diff
@@ -61,13 +61,11 @@ function mergeForEach<
       const lhsValue = getLHSValue(lhsItem) as any
       const rhsValue = getRHSValue(rhsItem) as any
 
-      if (lhsValue < rhsValue) {
-        return SORTING_ORDER.LHS_BEFORE_RHS
-      } else if (lhsValue > rhsValue) {
-        return SORTING_ORDER.LHS_AFTER_RHS
-      } else {
-        return SORTING_ORDER.EQUAL
-      }
+      if (lhsValue === rhsValue) return SORTING_ORDER.EQUAL
+      if (lhsValue < rhsValue) return SORTING_ORDER.LHS_BEFORE_RHS
+      if (lhsValue > rhsValue) return SORTING_ORDER.LHS_AFTER_RHS
+
+      throw new Error(`[@geoblink/lodash-mixins#mergeForEach] Found non-comparable values: ${lhsValue} on LHS and ${rhsValue} on RHS`)
```

Comment:
This might have impact where we use it. We have to be careful about this

Reply:
If we find this error then we were relying on the old bug which was actually an undefined/unexpected behaviour, I think it's better to fail even without us expecting it rather than giving false matches (this was happening in My Data wizard before and was a crazy thing to debug)

Reply:
I suppose